### PR TITLE
Update solutions/gold/sumdiv.mdx

### PR DESCRIPTION
**Pull request by Harshit1927**
The previous code was giving WA on the problem, linked correct code which gets AC and updated the explanation for the problem.

_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

**Comment by SansPapyrus683** [comment]
@brebenelmihnea uuuuhhh

**Review by brebenelmihnea — APPROVED** [review]
Nice work.

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/gold/sumdiv.mdx [hunk 1]
```diff
@@ -2,79 +2,131 @@
...
 	return res;
 }
 
-int inv(int x) {
-	// The Modular Inverse only exists if x % MOD != 0
-	if (x % MOD == 0) return 1;
-	if (x <= 1) { return x; }
-	return MOD - MOD / x * inv(MOD % x) % MOD;
+void ext(ll a, ll b, ll &x,
+         ll &y) {  // Extended euclidean gcd algorithm to find modular inverse
```
[the remark this answers is not in the document]
Comment (Harshit1927) [reply solutions/gold/sumdiv.mdx]: > weird formatting, put comment on top of function?

Is the formatting correct now?

**Comment by SansPapyrus683** [comment]
alright did some formatting myself, can you check to see if the code still works?

**Comment by Harshit1927** [comment]
> alright did some formatting myself, can you check to see if the code still works?

yes, it works

**Review by SansPapyrus683 — APPROVED** [review]

**Pull request merged**